webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueWriter.scala

```diff
@@ -267,9 +267,9 @@ class TechniqueWriter (
     }
   }
 
-  def runRudderLangTestLoop(techniqueBundleName: String): Unit = {
+  def runRudderLangTestLoop(techniqueBundleName: String, techniqueCategory: String): Unit = {
```

Comment:
You can pass a technique as parameter instead of the two subfields, and use bundle names and category from the technique on the right place inside this function.